src/search/pdbs/pattern_collection_generator_multiple_cegar.h

```diff
@@ -0,0 +1,41 @@
+#ifndef PDBS_PATTERN_COLLECTION_GENERATOR_MULTIPLE_CEGAR_H
+#define PDBS_PATTERN_COLLECTION_GENERATOR_MULTIPLE_CEGAR_H
+
+#include "pattern_generator.h"
+
+namespace options {
+class Options;
+}
+
+namespace utils {
+class RandomNumberGenerator;
+enum class Verbosity;
+}
+
+namespace pdbs {
+class PatternCollectionGeneratorMultipleCegar : public PatternCollectionGenerator {
+    // Options for the CEGAR algorithm.
+    const int max_pdb_size;
+    const int max_collection_size;
+    const bool wildcard_plans;
+    const double cegar_max_time;
+
+    // Options for this generator.
+    const utils::Verbosity verbosity;
+    std::shared_ptr<utils::RandomNumberGenerator> rng;
+    // We store the random seed for creating different RNG objects for CEGAR.
+    const int random_seed;
```

Comment:
You could instead store a second RandomNumberGenerator here (e.g.,  `std::shared_ptr<utils::RandomNumberGenerator> rng_for_cegar`) that you reuse in all calls to CEGAR.